### PR TITLE
fix: revalidateOnFocus not working on '< iOS 14' 

### PR DESCRIPTION
**Pull request by joshuakimDwan**
I found that revalidateOnFocus is not working on iOS 13. 
To fix, `window.addEventListener('visibilityevent', callback)` should be replaced by `document.addEventListener('visibilityevent', callback)`

Ref: https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event
image

### CODE REVIEW

File: src/use-swr.ts
Comment (huozhi) [inline src/use-swr.ts:58]: this will crash on deno and RN runtime cuz they don't have `document` as an accessible global reference.

maybe we can do like this? 🤔 

```js
if (!IS_SERVER) {
  const revalidate = () => ...
  
  const hasWindow = typeof window !== 'undefined'
  const hasDocument = typeof document !== 'undefined'
  
  if (hasDocument) {
     document.addEventListener(...)
  }
  if (hasWindow) {
     window.addEventListener(...)
     window.addEventListener(...)
  }
}
```

this might not be very elegant, feel free to share any other approaches

Comment (joshuakimDwan) [reply src/use-swr.ts:58]: I think your solution is intuitive and readable. I will apply it.

Comment (joshuakimDwan) [reply src/use-swr.ts:58]: By the way, I don't know RN development environment. 
So, I have one question. Does RN have document object? If not, I will add logic to deal with it.

Comment (joshuakimDwan) [reply src/use-swr.ts:58]: (+) I have another question on "this will crash on deno"
`deno` is not considered a server environment?
I mean `!IS_SERVER` doesn't mean `is not on deno environment`?

Comment (huozhi) [reply src/use-swr.ts:58]: oops, right deno is included in that server side condition.

Comment (joshuakimDwan) [reply src/use-swr.ts]: I see. I applied your solution on 5fd759c

File: src/use-swr.ts
Comment (huozhi) [inline src/use-swr.ts:69]: I think the 2 conditions doesn't have to be nested?

```
if (hasDocument) {
   // focus revalidators
}
if (hasWindow) {
   // other revalidators
}
```

Comment (joshuakimDwan) [reply src/use-swr.ts:69]: RN environment also have `document` object like the web? 
I think that if RN doesn't have `document` object we should use `window.addEventListener`

Comment (huozhi) [reply src/use-swr.ts:69]: nope. it has window object but not document. 
I double check again locally with a RN app. it shows that window doesn't have `addEventListener`. and we need to check addEventListener. looks like your first commit also works well and it's shorter

I'll approve it on my side if can be reverted to the previous commit. sorry for my mistake and thanks for the question.

Comment (joshuakimDwan) [reply src/use-swr.ts:69]: Aha. Is this logic ONLY for web, not RN?

Comment (huozhi) [reply src/use-swr.ts]: yes, swr supports browser runtime by default. since react is been ported to different rumtime, people are trying adopt swr along with react apps. I was wondering if it's possible to make swr more flexible to be adopted among runtimes. seems not easy.

**Review by huozhi — APPROVED** [review]

**Comment by mikevercoelen** [comment]
After merging this fix, it's now crashing when debugging react native and giving:

`document is not defined`

**Pull request merged**
